src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4726,6 +4750,30 @@ int wm_vuldet_set_agents_info(agent_software **agents_software, update_node **up
         } else if (strcasestr(os_name, vu_feed_ext[FEED_WIN])) {
             agent_dist_ver = wm_vuldet_decode_win_os(os_name);
             agent_dist = FEED_WIN;
+        } else if (strcasestr(os_platform, vu_feed_ext[FEED_MAC])) {
+            char *version = NULL;
+            w_strdup(agent_version, version);
+
+            if (agent_version) {
+                strtok(agent_version, "v");
+                agent_major=strtok(NULL, ".");
+                agent_minor=strtok(NULL, ".");
+                if (agent_major == NULL) {
+                    mdebug1("Cannot parse agent_major from version agent");
```

Comment:
Join these two conditions into a single log:
```
if (!agent_major || !agent_minor) {
  mterror(WM_VULNDETECTOR_LOGTAG, "Error reading Wazuh version for agent X"
}
```
Also, it would be better to create a log template at `error_messages.h`. See other examples.